src/components/Navigator/NavigatorCard.vue

```diff
@@ -316,6 +316,34 @@ export default {
         return titleMatch && tagMatch && hasAPIChanges && type !== TopicTypes.groupMarker;
       });
     },
+    /**
+     * Returns a Set of all nodes that match a filter, along with their parents.
+     * @returns NavigatorFlatItem[]
+     */
+    filteredChildrenUpToRootSet: ({ filteredChildren, getParents }) => new Set(
+      filteredChildren.flatMap(({ uid }) => getParents(uid)),
+    ),
+    /**
+     * This generates a map of all the nodes we are allowed to render at a certain time.
+     * This is used on both toggling, as well as on navigation and filtering.
+     * @return {Object.<string, NavigatorFlatItem>}
+     */
+    renderableChildNodesMap({ filteredChildrenUpToRootSet, childrenMap, hasFilter }) {
+      if (!hasFilter) return childrenMap;
+      const all = [];
+      // create a set of all matches and their parents
+      filteredChildrenUpToRootSet.forEach((current) => {
+        // if it has no children, just add it
+        if (!current.childUIDs.length) return all.push(current);
```

Comment:
This is probably working the way that you expect, but it's a little confusing to read since the `forEach` isn't expected to actually return anything, and you're possibly just using the returns to break the control flow(?). Also, `Array.push` returns a count which adds to the confusing way this reads, to me at least.

It might be helpful to rewrite this either using `Array.reduce` with `Array.concat` to add/return items or just a simple for loop without return statements for imperative actions like `Array.push`.

Or am I maybe just misunderstanding something here?

Reply:
Yeah it was to break to flow. Refactored it a bit. I cant use `reduce`, because I am looping a `Set` and I wanted to not convert it back to an array again.